src/unistd.rs

```diff
@@ -2136,3 +2136,201 @@ mod setres {
         Errno::result(res).map(drop)
     }
 }
+
+#[derive(Debug, Clone, PartialEq)]
+pub struct Passwd {
+    /// username
+    pub name: CString,
```

Comment:
Is it important to mark these all as `pub`?

Reply:
I think so. . . the whole point of these functions is to get data from this struct.

Reply:
One thing that other Nix modules do is create a Newtype to wrap the `libc` struct, and define accessors for that.  See for example `nix::sys::Kevent` or `nix::sys::UtsName`.  I think that pattern would work well here.  You'd do something like:
```rust
pub struct Passwd(libc::passwd);
impl Passwd {
    pub fn name(&self) -> &str {
        ...
    }
}
```

Reply:
@asomers That seems very Java-like to me and not necessary for these structs. Furthermore, some day I may also add `putpwent`, which would require a user to build this struct. `putpwent` is basically how the shell utilitty `adduser` works. http://man7.org/linux/man-pages/man3/putpwent.3.html

Reply:
Accessor functions may seem unnecessary to you, but converting C's `struct passwd` to a different layout seems unnecessary to me.  As long as we're dealing with C and Rust datatypes that have incompatible layouts, we're going to need some kind of conversion.  I like the accessor method version better, because that way we won't waste time converting fields that the user doesn't care about.  We'll probably save RAM too.